packages/hackmode/lib/index.js

```diff
@@ -40,87 +37,31 @@ const log = tweetlog('@hackmode@')
  * - `webproxy.wssAdvertise` {uri|"auto"} - Cannot be paired with `wssRelayPeers`. Sets up this node to be directly connectable at this address. Special case if set to `"auto"` will pick the first public NIC binding... allowing for os-assigned ports.
  * - `webproxy.wssRelayPeers` {array<uri>} - Cannot be paired with `wssAdvertise`. Uri array of relay peers to connect through. (currently you can only specify 1). Use this if behind a NAT, all messages will be routed through the peer specified here.
  */
-class N3hHackMode extends AsyncClass {
+class N3hHackMode extends N3hMode {
```

Comment:
I'm a little wary of this OO pattern, since sometimes it damages maintainability making it hard to know which base class to inspect for problems. Def glad to consolidate some of the duplication, we'll just have to keep an eye on it, if things get confusing we could switch to a composition pattern or something.

Reply:
Yeah, also not my favorite, but it was the easiest thing for me to pull off with javascript :p